### PR TITLE
Update swarm ID tag to const in flavor SPI

### PR DESCRIPTION
**Pull request by kaufers**
Similar to https://github.com/docker/infrakit/pull/810

### CODE REVIEW

File: pkg/spi/flavor/spi.go [hunk 1]
```diff
@@ -7,6 +7,11 @@ import (
 	"github.com/docker/infrakit/pkg/types"
 )
 
+const (
+	// SwarmIDTag is the name of the tag that contains the swarm ID
+	SwarmIDTag = "swarm-id"
```
Comment (chungers) [inline pkg/spi/flavor/spi.go]: Would you mind renaming this to `ClusterIDTag`?  

At the level of SPI, which is agnostic to *what* flavor it is, seeing a reference to Swarm feels strange.    I think it's totally reasonable to have something generic like a "cluster" or "stack" ID.  Indeed we already have the concept of stack elsewhere in the code base (e.g. the manager manages a "stack")

Comment (kaufers) [reply pkg/spi/flavor/spi.go]: @chungers sure, I'm fine with the constant var name. How about the value of the tag? Should it also change from `swarm-id` to `cluster-id`? Or, to be more consistent with other tags, should it be `swarm.id` or `cluster.id`?

Comment (chungers) [reply pkg/spi/flavor/spi.go]: How about `infrakit.cluster.id` ?

I made a similar comment on #836  -- I think it'd be useful to follow a convention for tags that are created by infrakit.

Comment (kaufers) [reply pkg/spi/flavor/spi.go]: Sure, I like that, it'll be:
```
ClusterIDTag = "infrakit.cluster.id"
```
Thanks for the review!

**Pull request merged**